feat: Allow resolving references after overlay application

jgraeger opened the pull request:
When working with overlays, oapi-codegen doesn't support external references, even though they are supported for the same file without overlays.

This PR adds a new setting `resolve-refs` to the OverlayOutput configuration. If enabled, the external references are resolved as for a normal file (against the filepath of the spec).

jamietanna commented:
Interesting - would you mind adding an example of this to `internal/test`?

I've not hit this before, but would be good to see what we're missing

jamietanna commented on configuration-schema.json:
```diff
@@ -209,6 +209,11 @@
+            "resolve-refs": {
+              "type": "boolean",
+              "description": "ResolveRefs enables reference resolving relative to the base path of the input file after the overlay is applied.",
+              "default": "false"
```
I appreciate you've done the work for wiring this in - I'd say this should be default behaviour

```suggestion
              "default": "true"
```

(I'd argue we could also not make it configurable at all, right now)

jgraeger commented:
@jamietanna Thank you for the swift review! I agree and removed config flag and enabled it by default. I also added an example to `internal/test` for overlay.

Apparently the issue only happens if the spec is not in the directory oapi-codegen is invoked from.  Therefore i have a subdirectory `spec/`. Running `go generate` after setting [ResolveRefs](https://github.com/oapi-codegen/oapi-codegen/blob/7c3f8c39bbfbb9017ef9c180c5202aa8883dfba0/cmd/oapi-codegen/oapi-codegen.go#L286) to false reproduces the error.

jgraeger commented:
@jamietanna Is the example sufficient or do you need more information? :)

jamietanna commented:
Thanks very much for this @jgraeger - this was a good catch, and I'm classing this as a bug, not a feature 😅

I've pushed a couple of tweaks - will wait for CI then merge

Main tweaks:

- no longer making it opt-in, and instead default behaviour
- moving the test to a different directory - you weren't expected to know this, and this is more my personal preference, what you'd done was absolutely great - thank you!

pixelsoccupied commented:
Tried this fix on my side and it works well! Is there any ETA for this release @jamietanna? Thanks!
